### PR TITLE
Centralize how we launch activities

### LINKED ISSUE #2486: Centralize how activities are launched
We would like to centralize how the activities are launched using the new `ActivityLauncher` class within the `viewpager-main-activity` branch, so that it'll be easier to maintain in the future and will give us the ability to add animations and so on from one central place.

### PR DESCRIPTION
**Pull request by oguzkocer**
This is the first pass at centralizing how we launch activities throughout the app. I suspect that there will be some other activities we would like to move to `ActivityLauncher` and some of them might be better kept in specific classes. Although some of the decisions were clear, I think some of the refactor is just personal preference. That's why I wanted to open up the PR early, so we have a place to discuss what to move, method signatures etc.

Here are a few things that might be valuable to discuss:

* In `CommentDetailFragment` inside the `editComment` method we open `EditCommentActivity` however there is a specific comment about not using the `getActivity().startActivityForResult()`, so I didn't move it to `ActivityLauncher`. I've looked into the issue in the comment but I didn't see if it's fixed yet or not.
* I think it makes sense to keep some of the helper classes such as `WordPressMediaUtils`, `WPWebViewActivity` & `ReaderActivityLauncher`. If we moved everything to `ActivityLauncher` it would be kind of a mess. On the other hand, I didn't feel the need to create another helper class so far, but that's something we can consider if the file gets too big to manage.
* This is something a little unrelated to the centralizing issue but I came across `areThemesAccessible` in `ThemeBrowserActivity` which basically opens the current posts page if the themes page is not accessible. Would it make sense to open the "My Site" page here instead? Also, maybe it would be better to do an accessibility check within the `ActivityLauncher` instead of `ThemeBrowserActivity` and just show a message to the user that it's not accessible and not change the page at all?
* I am feeling uneasy about not null-checking some of the parameters in `ActivityLauncher`. @nbradbury: I know we talked about this on Slack, but I am afraid that some valid calls will be made to `ActivityLauncher` and we should be silently failing them instead of letting the app crash to fix it later on. Would you like to re-consider this as well? I want to make sure we know all the down-sides of this before merging it in.

Please feel free to propose any changes as I was unsure on more than a couple things. Thanks in advance!

Closes #2486.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 1]
```diff
@@ -38,16 +49,21 @@ public static void viewCurrentSite(Context context) {
...
-        intent.putExtra(StatsActivity.ARG_LOCAL_TABLE_BLOG_ID, blog.getLocalTableBlogId());
+        intent.putExtra(StatsActivity.ARG_LOCAL_TABLE_BLOG_ID, blogLocalTableId);
         context.startActivity(intent);
     }
 
     public static void viewCurrentBlogPosts(Context context) {
+        ActivityLauncher.viewCurrentBlogPosts(context, true);
+    }
+
+    public static void viewCurrentBlogPosts(Context context, boolean animated) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: We can get rid of the `animated` parameter - that was only necessary when there was a drawer.

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 2]
```diff
@@ -108,7 +118,45 @@ public static void addNewBlogPostOrPage(Context context, Blog blog, boolean isPa
...
+            intent.putExtra(SSLCertsViewActivity.CERT_DETAILS_KEYS, lastFailureChainDescription);
+            context.startActivity(intent);
+        } catch (GeneralSecurityException e) {
+            AppLog.e(AppLog.T.API, e);
+        } catch (IOException e) {
+            AppLog.e(AppLog.T.API, e);
+        }
+    }
+
+    public static void viewSettings(Activity activity) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Let's add `ForResult` to the names of routines that call `startActivityForResult` (ex: `viewSettingsForResult`) to make them more self-documenting.

**Comment by nbradbury** [comment]
* Good call on 'CommentDetailFragment'. That's still an issue, so we should leave that code as is.
* Agreed about keeping the existing helper classes.
* Handling that `areThemesAccessible` check in `ActivityLauncher` and not changing the activity makes the most sense to me. 
* Regarding the null checking, I'd say do what you feel most comfortable with.

**Comment by oguzkocer** [comment]
@nbradbury I made the changes we have discussed. I have also merged the latest `feature/viewpager-main-activity`, so it's all up to date.

For the null check, I ended up just adding it as a silent fail for now. I feel it's better than the app crashing. Maybe we can add something a little more advanced later on. A simple helper class which is used throughout the app where we can do some assertions would be pretty nice. I can also add some logs when the null check fails if you think it'll be helpful. In normal circumstances, I don't think those methods will ever get hit with null parameters, but I thought a little defensive programming was warranted here.

I also decided to make the `isAccessible` method static in `ThemeBrowserActivity` so I can use that in `MySiteFragment` to hide the relevant section in the UI. I think this turned out to be the best solution overall.

With these changes, I think the branch is ready for a second review. I hope I didn't miss anything! Thanks again for your help so far :)

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**